File: src/day01/sonar.py
with open("input.txt") as f:
    lines = [int(s) for s in f.readlines()]

print(f"Part 1: {len([x for i, x in enumerate(lines) if x > lines[i - 1]])}")
print(f"Part 2: {sum(a < b for a, b in zip(lines, lines[3:]))}")

File: src/day01/Sonar.kt
package day01

import java.io.File

private val input = File("src/day01/input.txt").readLines().map { it.toInt() }

fun main() {
    println("Part 1: ${input.zipWithNext { a, b -> b > a }.sumOf { it.compareTo(false) }}")
    // Use the fact that A + B + C > B + C + D if D > A
    println("Part 2: ${input.zip(input.drop(3)).filter { (a, b) -> b > a }.size}")
}